go/private/go_repository.bzl

```diff
@@ -13,26 +13,34 @@
 # limitations under the License.
 
 def _go_repository_impl(ctx):
-  if ctx.attr.commit and ctx.attr.tag:
-    fail("cannot specify both of commit and tag", "commit")
-  if ctx.attr.commit:
-    rev = ctx.attr.commit
-  elif ctx.attr.tag:
-    rev = ctx.attr.tag
-  else:
-    fail("neither commit or tag is specified", "commit")
-
-  if ctx.attr.url:
+  if ctx.attr.urls:
+    disallowed = []
     # explicit source url
     if ctx.attr.vcs:
-      fail("cannot specify both of vcs and url")
+      disallowed.append("vcs")
+    if ctx.attr.commit:
+      disallowed.append("commit")
+    if ctx.attr.tag:
+      disallowed.append("tag")
+    if disallowed:
+      fail("cannot specify and of %s if url or urls is specified", "url")
     ctx.download_and_extract(
         url = ctx.attr.url,
```

Comment:
Can remove this argument, since ctx.attr.url shouldn't exist anymore.

`urls` is undocumented in `download_and_extract`. Any idea when it was added? Will this break our compatibility with Bazel 0.4.4?